great_expectations/datasource/new_new_new_datasource.py

```diff
@@ -0,0 +1,67 @@
+#!!! Rename this
+from typing import Dict, List
+from great_expectations.core.batch import Batch
+from great_expectations.datasource.base_data_asset import BaseDataAsset
+from great_expectations.datasource.misc_types import NewBatchRequestBase
+from great_expectations.types import DictDot
+from great_expectations.validator.validator import Validator
+
+
+class NewNewNewDatasource:
+
+    def __init__(
+        self,
+        name,
+    ):
+        #!!! Trying this on for size
+        # experimental-v0.15.1
+        # warnings.warn(
+        #     "\n================================================================================\n" \
+        #     "NewNewNewDatasource is an experimental feature of Great Expectations\n" \
+        #     "You should consider the API to be unstable.\n" \
+        #     "\n" \
+        #     "You can disable this warning by calling: \n" \
+        #     "from great_expectations.warnings import GxExperimentalWarning\n" \
+        #     "warnings.simplefilter(action=\"ignore\", category=GxExperimentalWarning)\n" \
+        #     "\n" \
+        #     "If you have questions or feedback, please chime in at\n" \
+        #     "https://github.com/great-expectations/great_expectations/discussions/DISCUSSION-ID-GOES-HERE\n" \
+        #     "================================================================================\n",
+        #     GxExperimentalWarning,
+        # )
+
+        self._name = name
+        self._assets = DictDot()
+
+    def add_asset(self, name:str, *args, **kwargs) -> BaseDataAsset:
+        raise NotImplementedError
+
+    def rename_asset(self, old_name: str, new_name: str) -> None:
+        if not isinstance(new_name, str):
+            raise TypeError(f"new_name must be of type str, not {type(new_name)}.")
+
+        if new_name in self._assets.keys():
+            raise KeyError(f"An asset named {new_name} already exists.")
+
+        self._assets[new_name] = self._assets.pop(old_name)
+        asset = self._assets[new_name]
+        asset.set_name(new_name)
+
+        return asset
+
+    def list_asset_names(self) -> List[str]:
```

Comment:
let's just get assets here; we are in a runtime environment and can use objects over keys

Reply:
I'm not sure what you mean. In this prototype, NewNewNewDatasource also supports and `assets` property. Is that what you're looking for?

I think it's important to have a convenience method that can return a list of asset names as strings. Hence: `list_asset_names`.

(`get_asset_name_list` would also work, if we make the style guide decision that we don't like `list_` as a verb.)